### PR TITLE
[TEST] `illegal_argument_exception` when downsampling all indexes in serverless

### PR DESCRIPTION
**Pull request by kkrik-es**
Downsampling wildcard and _all indexes raises different errors in
stateful (security disabled) vs serverless (security enabled):

Stateful:
```
[{type=index_not_found_exception, reason=no such index [test*], index_uuid=_na_, index=test*, stack_trace=[test*] org.elasticsearch.index.IndexNotFoundException: no such index [test*]\n\tat org.elasticsearch.xpack.downsample.TransportDownsampleAction.masterOperation(TransportDownsampleAction.java:205)
```

Serverless:
```
[{type=illegal_argument_exception, reason=the action indices:admin/xpack/downsample does not support wildcards; the provided index expression(s) [test*] are not allowed, stack_trace=org.elasticsearch.ElasticsearchException$1: the action indices:admin/xpack/downsample does not support wildcards; the provided index expression(s) [test*] are not allowed
```

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
Downsampling does not support using multiple indices. Doing it on multiple indices relies on ILM (or DSL) to do it one by one.

**Comment by kkrik-es** [comment]
> Downsampling does not support using multiple indices. Doing it on multiple indices relies on ILM (or DSL) to do it one by one. FOr some reason the exception thrown is different now...the reason the test has 'index_not_found' is that is tries to read a test whose name is 'test*' without doing wildcard expansion. I think before changing the test we need to understand why now we have the illegal argument exception. Do you have a stack trace?

Good point, Salvatore. It seems like DSL returns a different error here, compared to ILM. They both fail, which is a good thing :)

**Comment by salvatore-campagna** [comment]
Also I don't think this test uses DSL to run downsampling. My understanding is that it calls the (restricted) Rest API...which is possible probably because the test runs with higher-then-normal privileges.

File: x-pack/plugin/downsample/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/downsample/10_basic.yml [hunk 1]
```diff
@@ -445,7 +445,7 @@ setup:
           index.blocks.write: true
 
   - do:
-      catch: /index_not_found_exception/
+      catch: /index_not_found_exception|illegal_argument_exception/
```
Comment (martijnvg) [inline x-pack/plugin/downsample/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/downsample/10_basic.yml:448]: Maybe add a comment on top of this api call to explain why we either expect of the two exceptions?

**Review by martijnvg — APPROVED** [review]
One minor comment. LGTM otherwise.

The reason a different error is returned isn't because of serverless, but because of the fact that security is enabled. The way indices expressions with wildcards are expanded is different  with security enabled or disabled.

File: x-pack/plugin/downsample/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/downsample/10_basic.yml (hunk 1 above)
Comment (kkrik-es) [reply x-pack/plugin/downsample/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/downsample/10_basic.yml:450]: Done.

**Pull request merged**